bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/image/ImageCatalogClient.scala

```diff
@@ -0,0 +1,266 @@
+
+  def isNearby(query: ImageSearchQuery): Boolean =
+    catalog === query.catalog && isNearby(query.coordinates)
+
+  def isNearby(c: Coordinates): Boolean =
+    coordinates.angularDistance(c) <= maxDistance
+}
+
+object ImageSearchQuery {
+  /** @group Typeclass Instances */
+  implicit val equals: Equal[ImageSearchQuery] = Equal.equalA[ImageSearchQuery]
+
+  /** Harcoded value of the max FoV for any Gemini instrument */
+  val maxDistance: Angle = Angle.fromArcmin(4.5)
+
+  implicit class DeclinationShow(val d: Declination) extends AnyVal {
+    def toFilePart: String = Declination.formatDMS(d, ":", 2)
+  }
+
+  implicit class RightAscensionShow(val a: RightAscension) extends AnyVal {
+    def toFilePart: String = a.toAngle.formatHMS
+  }
+
+  implicit class CoordinatesShow(val c: Coordinates) extends AnyVal {
+    def toFilePart: String = s"ra_${c.ra.toFilePart}_dec_${c.dec.toFilePart}"
+  }
+
+  implicit class AngularSizeShow(val s: AngularSize) extends AnyVal {
+    def toFilePart: String = f"w_${s.ra.toArcsecs.toInt}_h_${s.dec.toArcsecs.toInt}"
+  }
+}
+
+/**
+  * Image in the file system
+  */
+case class ImageInFile(query: ImageSearchQuery, file: Path, fileSize: Long) {
+
+  /**
+    * Tests if the image contains the coordinates parameter
+    */
+  def contains(c: Coordinates): Boolean = {
+    val ε = query.catalog.adjacentOverlap
+    // Convert everything to radians, calculations in Angle-space don't work due to range overflow
+    // Image size
+    val εφ = (query.size.halfDec.toRadians - ε.toRadians).max(0)
+    val ελ = (query.size.halfRa.toRadians  - ε.toRadians).max(0)
+
+    // target coordinates
+    val φ = toRadians(c.dec.toDegrees)
+    val λ = c.ra.toAngle.toRadians
+
+    // image coordinates
+    val φ0 = toRadians(query.coordinates.dec.toDegrees)
+    val λ0 = query.coordinates.ra.toAngle.toRadians
+
+    val θ = cos(φ0)
+
+    // Distance
+    val Δφ = abs(φ - φ0)
+    val Δλ = abs(λ - λ0)
+
+    Δφ <= εφ && Δλ*θ <= ελ
+  }
+}
+
+object ImageInFile {
+  /** @group Typeclass Instances */
+  implicit val equals: Equal[ImageInFile] = Equal.equalA[ImageInFile]
+
+  val fileRegex: Regex = """img_(.*)_ra_(.*)_dec_(.*)_w_(\d*)_h_(\d*)\.fits.*""".r
+
+  /**
+    * Decode a file name into an image entry
+    */
+  def entryFromFile(file: File): Option[ImageInFile] = {
+    file.getName match {
+      case fileRegex(c, raStr, decStr, w, h) =>
+        for {
+          catalog <- ImageCatalog.byId(c)
+          ra      <- Angle.parseHMS(raStr).map(RightAscension.fromAngle).toOption
+          dec     <- Angle.parseDMS(decStr).toOption.map(_.toDegrees).flatMap(Declination.fromDegrees)
+          width   <- Angle.fromArcsecs(w.toInt).some
+          height  <- Angle.fromArcsecs(h.toInt).some
+        } yield ImageInFile(ImageSearchQuery(catalog, Coordinates(ra, dec), AngularSize(width, height)), file.toPath, file.length())
+      case _                           => None
```

Comment:
it means that somebody put a non-matching file name in the cache, while not common it cannot be discarded. Those should be just ignored. It would also let us evolve the naming schema if needed